From 858da959b726498e3e49a060279bbb24ad1af9cb Mon Sep 17 00:00:00 2001
From: Marek Vasut <marex@denx.de>
Date: Fri, 5 May 2023 05:35:20 +0200
Subject: [PATCH 3/3] gbm: Default to XRGB8888 framebuffer

The ARGB8888 framebuffer format for base canvas makes little sense
as the base canvas is unlikely to be transparent and require alpha.
Use XRGB8888 framebuffer format which is more widely supported by
DRM drivers as base plane pixel format. In case a transparent base
plane is required, the command line --pixelformat switch can be used.

Upstream-Status: Submitted [https://github.com/ardera/flutter-pi/pull/326]
Signed-off-by: Marek Vasut <marex@denx.de>
---
 src/flutter-pi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/flutter-pi.c b/src/flutter-pi.c
index 659706b..61f1557 100644
--- a/src/flutter-pi.c
+++ b/src/flutter-pi.c
@@ -1600,7 +1600,7 @@ static int init_display(void) {
     flutterpi.gbm.surface = NULL;
     flutterpi.gbm.modifier = DRM_FORMAT_MOD_LINEAR;
     if (!flutterpi.gbm.format)
-        flutterpi.gbm.format = DRM_FORMAT_ARGB8888;
+        flutterpi.gbm.format = DRM_FORMAT_XRGB8888;
 
     flutterpi.gbm.surface = gbm_surface_create_with_modifiers(flutterpi.gbm.device, flutterpi.display.width, flutterpi.display.height, flutterpi.gbm.format, &flutterpi.gbm.modifier, 1);
     if (flutterpi.gbm.surface == NULL) {
-- 
2.39.2

